src/go/rpk/pkg/cli/group/describe.go

```diff
@@ -238,3 +259,47 @@ func printLagPerTopic(groups kadm.DescribedGroupLags) {
 		}
 	}
 }
+
+func regexGroups(adm *kadm.Client, expressions []string) ([]string, error) {
+	// Now we list all groups to match against our expressions.
+	groups, err := adm.ListGroups(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("unable to list groups: 	%w", err)
+	}
+
+	return regexListedGroups(groups.Groups(), expressions)
+}
+
+func regexListedGroups(groups, expressions []string) ([]string, error) {
```

Comment:
Thanks for splitting these 2 functions, can you write a unit test for this function, please?

Reply:
Addressed.